openvpn/version.go

```diff
@@ -0,0 +1,38 @@
+package openvpn
+
+import (
+	"os/exec"
+	"syscall"
+)
+
+//Version function checks that openvpn is available, given path to openvpn binary
+func Version(openvpnBinary string) error {
+
+	process := NewProcess(openvpnBinary, "[openvpn binary check]")
+	if err := process.Start([]string{"--version"}); err != nil {
+		return err
+	}
+	cmdResult := process.Wait()
+
+	exitCode, err := extractExitCodeFromCmdResult(cmdResult)
+	//openvpn returns exit code 1 in case of --version paramter, if anything else is returned - treat as error
+	if err != nil || exitCode != 1 {
+		return err
```

Comment:
In case `err == nil` and `exitCode == 0`, you'll return `nil`, which means that this function passes. A separate error should be returned in that case, i.e.
```go
if err != nil {
  return err
}
if exitCode != 1 {
  return errors.New("Oh shit!")
}
```

Maybe this indicates a need for unit-test? ;)

Reply:
Yes and yes :) it could be unittested